### ISSUE TITLE
println needs a message

### ISSUE THREAD

**Issue by bassrock** [satyan/sugar#95]
I am getting the following error when trying to access the db. Any ideas?

```
java.lang.RuntimeException: An error occured while executing doInBackground()
            at android.os.AsyncTask$3.done(AsyncTask.java:300)
            at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
            at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
            at java.util.concurrent.FutureTask.run(FutureTask.java:242)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
            at java.lang.Thread.run(Thread.java:841)
     Caused by: java.lang.NullPointerException: println needs a message
            at android.util.Log.println_native(Native Method)
            at android.util.Log.e(Log.java:232)
            at com.orm.SugarDb.getDomainClass(Unknown Source)
            at com.orm.SugarDb.getDomainClasses(Unknown Source)
            at com.orm.SugarDb.createDatabase(Unknown Source)
            at com.orm.SugarDb.onCreate(Unknown Source)
            at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:252)
            at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:164)
            at com.orm.Database.getDB(Unknown Source)
            at com.orm.SugarRecord.find(Unknown Source)
            at com.orm.SugarRecord.find(Unknown Source)
            at com.dabsquared.android.spark.model.User.getUserFromObject(User.java:122)
            at com.dabsquared.android.spark.LoggingInFragment$2$1.doInBackground(LoggingInFragment.java:113)
            at com.dabsquared.android.spark.LoggingInFragment$2$1.doInBackground(LoggingInFragment.java:109)
            at android.os.AsyncTask$2.call(AsyncTask.java:288)
            at java.util.concurrent.FutureTask.run(FutureTask.java:237)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
            at java.lang.Thread.run(Thread.java:841)
```

**Comment by Harti** [comment]
Just in case someone stumbles across this thread during debugging research:

I had a slightly different issue - I was using `MyEntity.count(MyEntity.class, null, null)` in the `MyRecord()` **default constructor**.

Sugar states in the documentation that you should preserve the default constructor (example has it empty). However, it doesn't say why, and this cost me about three hours of heavy debugging. It is because, internally, Sugar tries to instantiate all `SugarRecord` classes in your project by using the default constructor (without calling `save()`) and storing these instances internally to look up your records later. 

If you try referencing the very same Entity within its default constructor, e.g. to implement a separate field for item sorting, the `SugarDb` class will end up throwing a `TargetInvocationException` at line 79, `SugarDb` will try to `catch` that (line 86) and print the error message, but instead (for some reason) it's causing a `NullPointerException` that will crash your application.

**Long story short:** do NOT use `SugarRecord` functions within the default constructor.


@satyan possibly something to add to the documentation and/or fix for 1.4?
